test/integration/app-document/pages/_document.js

```diff
@@ -0,0 +1,31 @@
+// _document is only rendered on the server side and not on the client side
+// Event handlers like onClick can't be added to this file
+
+// ./pages/_document.js
+import Document, { Head, Main, NextScript } from 'next/document'
+import flush from 'styled-jsx/server'
+
+export default class MyDocument extends Document {
+  static getInitialProps ({ renderPage }) {
+    const { html, head, errorHtml, chunks } = renderPage()
```

Comment:
Shall we get these from the `Document`. Like this:

```
static getInitialProps(info) {
  const initialProps = Document.getInitialProps(info);
  return {
    ...initialProps,
    customProperty: 'Hello World'
  } 
}
```

Then it's very simple and we don't need expose internal things like `renderPage`. And let's make sure we update this for `_app.js` and inside README as well.